src/components/animate-ui/effects/marquee-title.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useLayoutEffect, useRef, useState } from "react";
+
+interface MarqueeTitleProps {
+    title: string;
+    classNames: string;
+}
+
+export function MarqueeTitle(props: MarqueeTitleProps) {
+    const [titleAnimate, setTitleAnimate] = useState<boolean>(false);
+    const [boxWidth, setBoxWidth] = useState(150);
+    const titleDivRef = useRef<HTMLDivElement>(null);
+    const titleSpanRef = useRef<HTMLSpanElement>(null);
+
+    const handleTitleAnimate = (state: boolean) => {
+        setTitleAnimate(state);
+    };
+
+    useLayoutEffect(() => {
+        const handleResize = () => {
+            const width = titleDivRef.current?.getBoundingClientRect().width;
+            setBoxWidth(width ?? 150);
+        };
+
+        if (titleDivRef.current) {
+            const width = titleDivRef.current.getBoundingClientRect().width;
+            setBoxWidth(width);
+
+            window.addEventListener('resize', handleResize);
+        }
+
+        return () => window.removeEventListener('resize', handleResize);
+    }, []);
+
+    useEffect(() => {
+        handleTitleAnimate(textWrapAnimate());
+    }, [boxWidth]);
+
+    function textWrapAnimate() {
+        if (!titleSpanRef.current || !titleDivRef.current) {
+            return false;
+        }
+        if (props.title.length < 17) {
+            return false;
+        }
+        else {
+            const boxLength = Math.ceil((titleDivRef.current.getBoundingClientRect()).width);
+            if (boxLength > 170 && props.title.length < 19) {
+                return false;
+            }
+            return true;
+        }
+    };
```

Comment:
Just a small change before merging, move the `textWrapAnimate` to inside the so we don't have this as unstable reference.
Or even better, you can remove the `handleTitleAnimate` and call `setTitleAnimate` directly. Also, you need to add `props.title` to the dependency array